configure.ac

```diff
@@ -207,8 +207,14 @@ AC_ARG_ENABLE([32bit],
 
 if test "$ENABLED_32BIT" = "yes"
 then
-    AM_CFLAGS="$AM_CFLAGS -DNO_64BIT -DNO_CURVED25519_128BIT -m32"
-    AM_LDFLAGS="$AM_LDFLAGS -m32"
+    AM_CFLAGS="$AM_CFLAGS -DNO_64BIT -DNO_CURVED25519_128BIT -DNO_CHACHA_ASM -DWC_NO_CACHE_RESISTANT"
+
+    if test "$host_cpu" == "x86_64" || test "$host_cpu" == "aarch64"
+    then
+        AM_CFLAGS="$AM_CFLAGS -m32"
+        AM_LDFLAGS="$AM_LDFLAGS -m32"
+        enable_asm=no
```

Comment:
Why do we want wholesale removal of assembly in 32bit.  We have tfm optos for 32bit for example, with instructions in the README -> ChangeLog for how to get it to work.

Reply:
@toddouska : I believe the changelog section you are referring to is:

```
- ./configure builds default to fastmath option
    * Note, if on ia32 and building in shared mode this may produce a problem
      with a missing register being available because of PIC, there are at least
      6 solutions to this:
      1) --disable-fastmath , don't use fastmath
      2) --disable-shared, don't build a shared library
      3) C_EXTRA_FLAGS=-DTFM_NO_ASM , turn off assembly use
      4) use clang, it just seems to work
      5) play around with no PIC options to force all registers being open,
         e.g., --without-pic
      6) if static lib is still a problem try removing fPIE
```

The assembly support is disabled on x86_64 with the `-m32` option because it won't build with fastmath TFM assembly.

On my Mac with either `TFM_X86_64` or `TFM_X86` fails with:

```
In file included from wolfcrypt/src/tfm.c:51:
In file included from ./wolfcrypt/src/asm.c:1362:
wolfcrypt/src/fp_sqr_comba_8.i:50:4: error: invalid operand for instruction
   SQRADD(a[0],a[0]);
   ^
./wolfcrypt/src/asm.c:702:28: note: expanded from macro 'SQRADD'
     "mull  %%eax        \n\t"                            \
                           ^
<inline asm>:3:13: note: instantiated into assembly here
        addl  %eax,%cx     
                   ^~~
```

I've updated the configure.ac to only disable the assembly if fastmath is enabled. I also only do this for `x86_64` as aarch64 doesn't have this issue.

Please let me know if you have a better suggestion.

Reply:
Just because one compiler on one OS can't build TFM with assembly in 32bit mode on a 64bit machine (possibly with the wrong defines NO_64BIT ?) doesn't mean to me that we should force all assembly off, especially if there's a simple workaround.